gladier_kanzus/tools/plot.py

```diff
@@ -1,19 +1,39 @@
-from typing import List
 from gladier import GladierBaseTool, generate_flow_definition
 
-
-def ssx_plot(x_num_steps=0, y_num_steps=0, int_indices=[], plot_filename='composite.png'):
-    """
-    Plot the current list of ints so far. Data requires the following keys
-        * x_num_steps (int) X dimension for the plot
-        * y_num_steps (int) Y dimension for the plot
-        * int_indices (list of ints) list of int 'hits' to light up on the plot
-        * plot_name (path) full filename to save the plot
-    """
+def ssx_plot(**data):
     import numpy as np
     from matplotlib import pyplot as plt
+    import os
+    import json
+    import re
+    import glob 
+
+    data_dir = data['data_dir']
+    proc_dir = data['proc_dir']
+    upload_dir = data['upload_dir']
+    run_num = data['run_num']
+
+    beamline_file_name = f'beamline_run{run_num}.json'
+    beamline_file = os.path.join(data_dir, beamline_file_name)
+
+    with open(beamline_file, 'r') as fp:
+        beamline_metadata = json.load(fp)
+        user_input = beamline_metadata.get('user_input', {})
+    
+    x_num_steps= user_input.get('x_num_steps')
+    y_num_steps= user_input.get('y_num_steps')
+    plot_filename=os.path.join(upload_dir, 'composite.png')
+
+    int_indices=[]
+    for int_file in glob.glob(os.path.join(proc_dir,'int-*.pickle')):
+        int_file = os.path.basename(int_file)
+        int_match = re.match(r'int-\d+-\w+_\d+_(\d+).pickle', int_file)
+        if int_match:
+            int_index = int_match.groups()[0]
+            int_indices.append(int(int_index))    
```

Comment:
If possible, I think it would be better to keep this simple. The old signature had some major advantages, the function didn't need to be aware of the entire `**data` input. Instead it only required 4 generic parameters. Making it responsible for knowing where the beamline file is etc. makes it more brittle and easier to break if anything changes, for instance if the `x_num_steps` var changes within the beamline.json file.

Reply:
Let me ponder on this. I could make a hybrid that accepts both styles
With **data as it is written right now, it shares the exact same structure as all other funcx-files and you only need to deal with inputs on the client.

Reply:
I would aim for a method where the code wasn't copied between the funcx functions. For future maintenance, we'd need to edit all existing copies any time one of them changed.

Reply:
I agree. Just not sure how to implement that now without complicating the flow.